server/server.vala

```diff
@@ -390,6 +390,16 @@ namespace Frida.Server {
 				connection.unregister_object (registration_id);
 			}
 
+			private void schedule_idle (owned ScheduledFunc function) {
+				var client = this;
+				Idle.add(() => {
+					function (client);
```

Comment:
How about dropping the argument from the delegate and instead doing:

```Vala
client = null;
```

after calling `func`?